File: remove_bg.py
from PIL import Image
import sys

def remove_white_bg(input_path, output_path, threshold=220):
    try:
        img = Image.open(input_path)
        img = img.convert("RGBA")
        datas = img.getdata()

        newData = []
        for item in datas:
            # Check if pixel is white (close to 255, 255, 255)
            if item[0] >= threshold and item[1] >= threshold and item[2] >= threshold:
                newData.append((255, 255, 255, 0))
            else:
                newData.append(item)

        img.putdata(newData)
        img.save(output_path, "PNG")
        print("Successfully removed background")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    remove_white_bg(
        "c:\\Users\\adrad\\OneDrive\\Desktop\\mezi-health-pkg\\.png\\medcy_logo_high_res.png", 
        "c:\\Users\\adrad\\OneDrive\\Desktop\\mezi-health-pkg\\public\\medcy_logo.png"
    )

File: remove_bg.js
const Jimp = require('jimp');

async function removeWhiteBg() {
    try {
        const inputPath = 'c:\\Users\\adrad\\OneDrive\\Desktop\\mezi-health-pkg\\.png\\medcy_logo_high_res.png';
        const outputPath = 'c:\\Users\\adrad\\OneDrive\\Desktop\\mezi-health-pkg\\public\\medcy_logo.png';

        console.log('Loading image...');
        const image = await Jimp.read(inputPath);

        console.log('Processing pixels...');
        image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
            const r = this.bitmap.data[idx + 0];
            const g = this.bitmap.data[idx + 1];
            const b = this.bitmap.data[idx + 2];

            // If pixel is close to white, make it transparent
            if (r > 230 && g > 230 && b > 230) {
                this.bitmap.data[idx + 3] = 0;
            }
        });

        console.log('Saving image...');
        await image.writeAsync(outputPath);
        console.log('Successfully removed background and saved to public/medcy_logo.png');
    } catch (error) {
        console.error('Error processing image:', error);
    }
}

removeWhiteBg();
